builtin/providers/aws/resource_aws_alb_listener_rule.go

```diff
@@ -0,0 +1,252 @@
+						"field": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateAwsListenerRuleField,
+						},
+						"values": {
+							Type:     schema.TypeList,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsAlbListenerRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	elbconn := meta.(*AWSClient).elbv2conn
+
+	params := &elbv2.CreateRuleInput{
+		ListenerArn: aws.String(d.Get("listener_arn").(string)),
+		Priority:    aws.Int64(int64(d.Get("priority").(int))),
+	}
+
+	actions := d.Get("action").([]interface{})
+	params.Actions = make([]*elbv2.Action, len(actions))
+	for i, action := range actions {
+		actionMap := action.(map[string]interface{})
+		params.Actions[i] = &elbv2.Action{
+			TargetGroupArn: aws.String(actionMap["target_group_arn"].(string)),
+			Type:           aws.String(actionMap["type"].(string)),
+		}
+	}
+
+	conditions := d.Get("condition").([]interface{})
+	params.Conditions = make([]*elbv2.RuleCondition, len(conditions))
+	for i, condition := range conditions {
+		conditionMap := condition.(map[string]interface{})
+		values := conditionMap["values"].([]interface{})
+		params.Conditions[i] = &elbv2.RuleCondition{
+			Field:  aws.String(conditionMap["field"].(string)),
+			Values: make([]*string, len(values)),
+		}
+		for j, value := range values {
+			params.Conditions[i].Values[j] = aws.String(value.(string))
+		}
+	}
+
+	resp, err := elbconn.CreateRule(params)
+	if err != nil {
+		return errwrap.Wrapf("Error creating ALB Listener Rule: {{err}}", err)
+	}
+
+	if len(resp.Rules) == 0 {
+		return errors.New("Error creating ALB Listener Rule: no rules returned in response")
+	}
+
+	d.SetId(*resp.Rules[0].RuleArn)
+
+	return nil
+}
+
+func resourceAwsAlbListenerRuleRead(d *schema.ResourceData, meta interface{}) error {
+	elbconn := meta.(*AWSClient).elbv2conn
+
+	resp, err := elbconn.DescribeRules(&elbv2.DescribeRulesInput{
+		RuleArns: []*string{aws.String(d.Id())},
+	})
+	if err != nil {
+		if isRuleNotFound(err) {
+			log.Printf("[WARN] DescribeRules - removing %s from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return errwrap.Wrapf(fmt.Sprintf("Error retrieving Rules for listener %s: {{err}}", d.Id()), err)
+	}
+
+	if len(resp.Rules) != 1 {
+		return fmt.Errorf("Error retrieving Rule %q", d.Id())
+	}
+
+	rule := resp.Rules[0]
+
+	d.Set("arn", rule.RuleArn)
+	if priority, err := strconv.Atoi(*rule.Priority); err != nil {
+		return errwrap.Wrapf("Cannot convert rule priority %q to int: {{err}}", err)
+	} else {
+		d.Set("priority", priority)
+	}
+
+	actions := make([]interface{}, len(rule.Actions))
+	for i, action := range rule.Actions {
+		actionMap := make(map[string]interface{})
+		actionMap["target_group_arn"] = *action.TargetGroupArn
+		actionMap["type"] = *action.Type
+		actions[i] = actionMap
+	}
+	d.Set("action", actions)
```

Comment:
This is somewhat following convention - I'm not too worried about it here. In general though I'd agree we should formalise more of the policy about where we check errors on `d.Set()`